portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -1868,11 +1870,19 @@ public void includeLogin(PortalRenderContext rcontext, HttpServletRequest req,
 					loginUserDispName = Validator.escapeHtml(thisUser.getDisplayName());
 					loginUserFirstName = Validator.escapeHtml(thisUser.getFirstName());
 				}
+				
+				// check if current user is being impersonated (by become user/sutool)
+				String impersonatorEid = getImpersonatorEid();
+				if (!impersonatorEid.isEmpty())
+				{
+					message = rloader.getString("sit_return") + " " + impersonatorEid;
+				}
 
 				// check for a logout text override
-				message = StringUtils.trimToNull(ServerConfigurationService
-						.getString("logout.text"));
-				if (message == null) message = rloader.getString("sit_log");
+				if (message == null)
+				{
+					message = StringUtils.defaultIfBlank(ServerConfigurationService.getString("logout.text"), rloader.getString("sit_log"));
```

Comment:
Please just use the ServerConfigurationService.getString pattern with a default value.